### ISSUE TITLE
Can HexaPDF write a page as an image?

### ISSUE THREAD

**Issue by gbirchmeier** [gettalong/hexapdf#317]
(I apologize for using an issue to ask a question, but I didn't see in the docs any suggested forum to ask questions or seek help.)

Can I use HexaPDF to convert a document to a collection of image files, where each image is a whole page?

I've experimented quite a bit this morning, and dug through the docs, but I didn't find anything to fit this use case.

**Comment by gettalong** [comment]
No, rendering pages to a canvas is a feature that HexaPDF doesn't implement and will not implement in the foreseeable future.

There are a variety of tools that can do this, my favourite is `pdftocairo` from the poppler tools.

**Comment by gbirchmeier** [comment]
Thanks.  I did end up finding a solution with RMagick, but I was really attracted to the possibility of a pure-Ruby solution.  Oh well.

**Comment by gettalong** [comment]
Alas, I know of no pure Ruby solution for this.

As for RMagick: It works *most* of the time but the rendering of more complex PDFs is not really good from what I remember. `pdftocairo` on the other hand uses the same backend as the prominent open source PDF viewers Okular and Evince.

**Comment by gbirchmeier** [comment]
I'll keep that in mind in case my RMagick script falters.

My use case is just converting PDFs to CBZs (comic-book zip), because large PDFs get a little laggy on some comic-reader apps.  (Google's PDFs have each *panel* as a separate image, which surprised me when I first used HexaPDF to extract images.)

I did also try https://github.com/robflynn/pdftoimage which uses poppler, but failed to figure out how to install the "poppler_utils" that its README says it needs.

Thanks for the help.
